include/shark/Algorithms/DirectSearch/Operators/Hypervolume/HypervolumeSubsetSelection2D.h

```diff
@@ -0,0 +1,298 @@
+			if(f.eval(0)  > res) {
+				res = f.eval(0);
+				currentIndex = i;
+			}
+		}
+		front[currentIndex].selected = true;
+		//iterate backwards to reconstruct chosen indizes
+		for(auto pos = chosen.rbegin(); pos != chosen.rend(); ++pos){
+			currentIndex = (*pos)[currentIndex];
+			front[currentIndex].selected = true;
+		}
+	}
+	
+	template<typename Set>
+	std::vector<Point> createFront(Set const& points, double refX, double refY)const{
+		//copy points using the new reference frame with refPoint at (0,0). also store original index for later
+		std::vector<Point> front;
+		for(std::size_t i = 0; i != points.size(); ++i){
+			front.emplace_back(points[i](0) - refX, points[i](1) - refY,i);
+		}
+		std::sort(front.begin(),front.end());//sort lexicographically
+		//erase dominated points
+		auto newEnd = std::unique(front.begin(),front.end(),[](Point const& x, Point const& y){
+			return y.f2 >= x.f2;//by lexikographic sort we already have y.f1 >= x.f1
+		});
+		front.erase(newEnd,front.end());
+		return front;
+	}
+public:
+	/// \brief Executes the algorithm.
+	/// \param [in] points The set \f$S\f$ of points to select
+	/// \param [out] selected set of the same size as the set of points indicating whether the set is selected (1) or not (0)
+	/// \param [in] k number of points to select
+	/// \param [in] refPoint The reference point \f$\vec{r} \in \mathbb{R}^2\f$ for the hypervolume calculation, needs to fulfill: \f$ \forall s \in S: s \preceq \vec{r}\f$. .
+	template<typename Set, typename SelectedSet, typename VectorType >
+	void operator()( Set const& points, SelectedSet& selected, std::size_t k, VectorType const& refPoint){
+		SIZE_CHECK(points.size() == selected.size());
+		if(points.empty())
+			return;
+		SIZE_CHECK( points.begin()->size() == 2 );
+		SIZE_CHECK( refPoint.size() == 2 );
+		
+		for(auto&& s: selected)
+			s = false;
+		
+		std::vector<Point> front = createFront(points, refPoint(0), refPoint(1));
+		
+		//find the optimal set in the front. afterwards selected points have selected=true
+		hypSSP(front,k);
+		//mark selected points in the original front
+		for(Point const& point: front){
+			if(point.selected){
+				selected[point.index] = true;
+			}
+		}
+	}
+	
+	/// \brief Executes the algorithm.
+	///
+	/// This version does not use a reference point. instead the extreme points are kept which defines a reference point
+	/// \param [in] points The set \f$S\f$ of points to select
+	/// \param [out] selected set of the same size as the set of points indicating whether the set is selected (1) or not (0)
+	/// \param [in] k number of points to select, must be larger than 2
+	template<typename Set, typename SelectedSet, typename VectorType >
+	void operator()( Set const& points, SelectedSet& selected, std::size_t k){
+		SIZE_CHECK(points.size() == selected.size());
+		if(points.empty())
+			return;
+		SIZE_CHECK( points.begin()->size() == 2 );
+		SHARK_CHECK( k > 2, "[HypervolumeSubsetSelection2D] k must be larger than 2");
```

Comment:
>> +		}
>> +	}
>> +	
>> +	/// \brief Executes the algorithm.
>> +	///
>> +	/// This version does not use a reference point. instead the extreme points are kept which defines a reference point
>> +	/// \param [in] points The set \f$S\f$ of points to select
>> +	/// \param [out] selected set of the same size as the set of points indicating whether the set is selected (1) or not (0)
>> +	/// \param [in] k number of points to select, must be larger than 2
>> +	template<typename Set, typename SelectedSet, typename VectorType >
>> +	void operator()( Set const& points, SelectedSet& selected, std::size_t k){
>> +		SIZE_CHECK(points.size() == selected.size());
>> +		if(points.empty())
>> +			return;
>> +		SIZE_CHECK( points.begin()->size() == 2 );
>> +		SHARK_CHECK( k > 2, "[HypervolumeSubsetSelection2D] k must be larger than 2");
>
> this would be trivial, right? pick the edge points and be done. The algorithm can not really do its job as the set does not containa  single point that is actually picked by hypervolume.

Yes, it is a trivial corner case. But why should it assert? It should 
simply return the edge points (and succeed).